components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/RequestObjectValidatorUtil.java

```diff
@@ -245,4 +310,14 @@ public static boolean isSignatureVerified(SignedJWT signedJWT, Certificate x509C
             return false;
         }
     }
+
+    private static boolean isFapiConformant(String clientId) throws RequestObjectException {
+
+        try {
+            return OAuth2Util.isFapiConformantApp(clientId);
```

Comment:
The method throws both the errors as IdentityOAuth2Exception. Even in the other places where this method is used its handled as a system exception. 

https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/fa3398330a1bd8ea8ff7bd9fa3935ff808322b24/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/RequestObjectValidatorUtil.java#L113-L116

https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/fa3398330a1bd8ea8ff7bd9fa3935ff808322b24/components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java#L2707-L2716

https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/fa3398330a1bd8ea8ff7bd9fa3935ff808322b24/components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java#L1246-L1249

Reply:
In that case, I would say those other places are also wrong.

What happens there is the client sends a wrong client id and then we propagate that outside as a server exception. That means the client will not know that this is due to a client error. If the proper exception was propagated the client can handle this error and rectify it. Plus the current way we add an unnecessary server exception that the server can never fix.

Reply:
To get the service provider we have used the method getServiceProvider using client id in OAuth2Util class [1]. In this method, it catches the two exceptions and throw as a IdentityOAuth2Exception[2]. Hence both the client and system exceptions are thrown using the same exception, and we were unable to identify them separately from the isFapiConformantApp method. 
Since this is a public method and used in several places we can't change the signature to throw both errors from that method without catching and throwing a single error. Is there a better approach we can follow here to identify the two exceptions separately?

[1] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java#L3579
[2] https://github.com/wso2-extensions/identity-inbound-auth-oauth/blob/master/components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java#L3579

Reply:
fixed with 2d38d538f3c599b89971a9484e6fd7f59be23982